### PR TITLE
Fix client IPs being broken on Python 3

### PR DESCRIPTION
**Pull request by hawkowl**
Fixes https://github.com/matrix-org/synapse/issues/3893

### CODE REVIEW

File: synapse/storage/_base.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def _do_execute(self, func, sql, *args):
                 sql, *args
             )
         except Exception as e:
-            logger.debug("[SQL FAIL] {%s} %s", self.name, e)
+            logger.error("[SQL FAIL] {%s} %s", self.name, e)
```
Comment (erikjohnston) [inline synapse/storage/_base.py]: Please don't, as this can get really quite spammy iirc

File: tests/utils.py [hunk 2]
```diff
@@ -237,20 +239,37 @@ def is_threepid_reserved(threepid):
...
+                    try:
+                        cur.execute("DROP DATABASE IF EXISTS %s;" % (test_db,))
+                        db_conn.commit()
+                        dropped = True
+                    except psycopg2.OperationalError as e:
+                        warnings.warn(
+                            "Couldn't drop old db: " + str(e), category=UserWarning
+                        )
+                        time.sleep(0.5)
+
```
Comment (erikjohnston) [inline tests/utils.py]: Can you add quick comment why this is necessary?

File: synapse/storage/_base.py (hunk 1 above)
Comment (hawkowl) [reply synapse/storage/_base.py]: From the ticket from @richvdh :

>  if update_client_ips is failing, it needs to log at ERROR or above.

Do we not want to know when SQL fails?

Comment (erikjohnston) [reply synapse/storage/_base.py]: Well we raise and handle it later. There are a bunch of failure modes that are non-error case, but instead get retried: https://github.com/matrix-org/synapse/blob/master/synapse/storage/_base.py#L249-L280

I'd really expect the exception to bubble up all the way to the client IP code, so I wonder if we're just swallowing it there?

**Pull request merged**